src/System.Xml.XDocument/src/System/Xml/Linq/Extensions.cs

```diff
@@ -374,9 +374,10 @@ static IEnumerable<XAttribute> GetAttributes(IEnumerable<XElement> source, XName
                     XContainer c = root;
                     while (true)
                     {
-                        if (c != null && c.content is XNode)
+                        XNode content = c != null ? c.content as XNode : null;
```

Comment:
If c is null, this appears to be adding an extra null check; before it would do the comparison and fall through to the else, but now it's setting content to null and then content is checked for whether it's null.  I don't know if that actual check would actually have an impact on perf, though.